shttps/Connection.cpp

```diff
@@ -471,21 +470,24 @@ namespace shttps {
             }
             else if (method_in == "POST") {
                 _method = POST;
-                if (content_length > 0) {
-                    vector <string> content_type_opts = process_header_value(header_in["content-type"]);
-                    if (content_type_opts[0] == "application/x-www-form-urlencoded") {
-                        char *bodybuf = nullptr;
-                        if (_chunked_transfer_in) {
-                            char *chunk_buf;
-                            ChunkReader ckrd(ins);
-                            content_length = ckrd.readAll(&chunk_buf);
-                            if ((bodybuf = (char *) malloc((content_length + 1) * sizeof(char))) == nullptr) {
-                                throw Error(__file__, __LINE__, "malloc failed!", errno);
-                            }
-                            memcpy(_content, chunk_buf, content_length);
-                            free(chunk_buf);
+                vector <string> content_type_opts = process_header_value(header_in["content-type"]);
+                if (content_type_opts[0] == "application/x-www-form-urlencoded") {
+                    char *bodybuf = nullptr;
+                    if (_chunked_transfer_in) {
+                        char *chunk_buf;
+                        ChunkReader ckrd(ins, _server->max_post_size());
+                        content_length = ckrd.readAll(&chunk_buf);
+                        if ((bodybuf = (char *) malloc((content_length + 1) * sizeof(char))) == nullptr) {
+                            throw Error(__file__, __LINE__, "malloc failed!", errno);
                         }
-                        else {
+                        memcpy(bodybuf, chunk_buf, content_length);
+                        free(chunk_buf);
```

Comment:
Another idea: could the pointer to the buffer be a member variable in the `ChunkReader` class? Then `ChunkReader` could free the memory in its destructor. Then there would be no need for the caller to free it.